ghcide/src/Development/IDE/Plugin/Completions.hs

```diff
@@ -77,10 +77,11 @@ produceCompletions = do
               let env = hscEnv sess
               -- We do this to be able to provide completions of items that are not restricted to the explicit list
               res <- liftIO $ tcRnImportDecls env (dropListFromImportDecl <$> imps)
-              case res of
-                  (_, Just rdrEnv) -> do
+              inScope <- liftIO $ tcRnImportDecls env imps
+              case (res, inScope) of
+                  ((_, Just rdrEnv), (_, Just inScopeEnv)) -> do
                       let uri = fromNormalizedUri $ normalizedFilePathToUri file
-                      cdata <- liftIO $ cacheDataProducer uri env (ms_mod ms) rdrEnv imps parsedDeps
+                      cdata <- liftIO $ cacheDataProducer uri env (ms_mod ms) rdrEnv inScopeEnv imps parsedDeps
```

Comment:
rather than passing both `inScopeEnv` and `rdrEnv`, is there any way to pass just the difference?

Reply:
Thanks for your suggestion! I was going to use IntMap.merge, but I realized that `UniqFM` behind `OccEnv` is opaque. Since we don't need keys (`OccName`s) in `cacheDataProducer`, I extract elements and then use `getDiff` to mark if it's in scope.